custom/icds_reports/tasks.py

```diff
@@ -58,3 +64,52 @@ def move_ucr_data_into_aggregation_tables(date=None):
                 sql_to_execute = sql_file.read()
                 cursor.execute(sql_to_execute, {"date": date.strftime('%Y-%m-%d')})
             celery_task_logger.info("Ended icds reports update_daily_aggregate_table")
+
+
+@periodic_task(run_every=crontab(minute=0, hour=21), acks_late=True, queue='background_queue')
+def recalculate_stagnant_cases():
+    domain = 'icds-cas'
+    config_ids = [
+        'static-icds-cas-static-ccs_record_cases_monthly',
+        'static-icds-cas-static-ccs_record_cases_monthly_v2',
+        'static-icds-cas-static-ccs_record_cases_monthly_tableau_v2',
+        'static-icds-cas-static-child_cases_monthly',
+        'static-icds-cas-static-child_cases_monthly_v2',
+        'static-icds-cas-static-child_cases_monthly_tableau_v2',
+    ]
+
+    stagnant_cases = set()
+
+    for config_id in config_ids:
+        config, is_static = get_datasource_config(config_id, domain)
+        adapter = get_indicator_adapter(config)
+        case_ids = _find_stagnant_cases(adapter)
+        celery_task_logger.info(
+            "Found {} stagnant cases in config {}".format(len(case_ids), config_id)
+        )
+        stagnant_cases = stagnant_cases.union(set(case_ids))
+        celery_task_logger.info(
+            "Total number of stagant cases is now {}".format(len(stagnant_cases))
+        )
+
+    case_accessor = CaseAccessors(domain)
+    num_stagnant_cases = len(stagnant_cases)
+    current_case_num = 0
+    for case_ids in chunked(case_ids, 100):
+        current_case_num += len(case_ids)
+        cases = case_accessor.get_cases(case_ids)
+        for case in cases:
+            publish_case_saved(case, send_post_save_signal=False)
+        celery_task_logger.info(
+            "Resaved %d / %d cases".format(current_case_num, num_stagnant_cases)
+        )
+
+
+def _find_stagnant_cases(adapter):
+    stagnant_date = datetime.utcnow() - timedelta(days=45)
+    table = adapter.get_table()
+    query = adapter.get_query_object()
+    query = query.with_entities(table.columns.doc_id).filter(
+        table.columns.inserted_at <= stagnant_date
+    ).distinct()
+    return query.all()
```

Comment:
based on @sheelio's auto update rules that he had setup, he was using the case's server_modified_on timestamp to select cases that hadn't been modified in 45 days. it looks like this is selecting cases who had records inserted into the ucr table more than 45 days ago. @sheelio does that still fulfill your use case?

Reply:
This was intentional by me. If we go by server_modified_on we'll need to either change that when queueuing (which seems weird to me as it's comment says its the date of the last transaction) or recalculate all unupdated cases every night (could be a lot of cases)